### PR TITLE
WIP: Add `--every` option, change default behavior, resolve #46

### LINKED ISSUE #46: Delay for triggering
Hello!

`filewatcher` has an `--interval` option.

But it's only for files check.

I have a problem: massive files changes by git (for example), when many-many files are changed, or added, or anything else, and in the same time `filewatcher` is working — it's begin triggering `shell command` many-many times.

Using with `--restart`.

`--interval` doesn't help.

We need for a code (or option) about trigger delay: for example, if any (watched) file has been changed — trigger `shell command` and wait 100–300 ms (for example) before next triggering.

Or, may be, don't collect collection of changed files, and trigger only by first.

### PR DESCRIPTION
**Pull request by AlexWayfer**
Also:

*   Filewatcher lib now returns `changes` Hash(filename => event) into `watch` block.

*   Many refactorings, less RuboCop warnings.

*   Fix typos in README

### CODE REVIEW

**Comment by thomasfl** [comment]
Hi Alex, I am so sorry for taking so long to answer.

A similar js command line utility has a debounce option, that does almost the same as an every option.

For example using "--debounce 2000" option, means that if there is a lot of file system changes happening rapidly after each other, then wait for 2000 milleseconds of no changes in the filesystem before triggering a filesystem change event. In other words we want to specify how long the "silence" period with no events is before emitting an filewatcher change event. 

```
    -d, --debounce          Debounce timeout in ms for executing command [default: 400]
```
Source: https://github.com/kimmobrunfeldt/chokidar-cli

What do you think? Would you prefer "debounce" over "every"?

**Comment by AlexWayfer** [comment]
Hi Thomas. Thanks for your answer.

I saw a lot of command spawns after mass update files (change branches in the git, for example).

I thought it would help the option to delay calling these commands, if command execution is real-time for each modified file (for example, git can modify each file in 5ms), or if the update files has been delayed for several filewatcher checks (like `--interval 0.5s`, but filesystem update taken 2s, filewatcher receive updates 3–4 times, and there is helpful option `--wait 2s`; but there is easier to increase `--interval`).

But I did get a grasp of the code, and realized that in one filewatcher check command is executed for each modified file. Accordingly, if you do `git checkout another_branch` which changes to 20 files at once — will be performed the command (restart the server, for example) 20 times in a row without delay ([there is a cycle](https://github.com/thomasfl/filewatcher/blob/master/lib/filewatcher.rb#L34), [there is no delay](https://github.com/thomasfl/filewatcher/blob/master/lib/filewatcher.rb#L46), [there is break for each file](https://github.com/thomasfl/filewatcher/blob/master/lib/filewatcher.rb#L124)).

So I thought (so far) enough option of one-time command execution for any number of changed files.

It is rather intuitive behavior for the CLI, I made it the default ([reference](https://github.com/thomasfl/filewatcher/issues/46#issuecomment-281060699)). To the old scheme, you can return with the option `--every` (name can be changed) ([reference](https://github.com/thomasfl/filewatcher/issues/46#issuecomment-281106500)). Ruby Lib now always return list (Hash) of changes (file => event). CLI get this Hash and iterate on it if option has been specified.

File: README.md [hunk 1]
```diff
@@ -23,16 +23,16 @@ In Windows:
...
 
 ```
 $ [sudo] gem install filewatcher
 ```
 
 ## Command line utility
 
-Filewatcher scans the filesystem and execute shell commands when files are
-updated, added, renamed or deleted.
+Filewatcher scans the filesystem and execute shell command when files are
```
Comment (thomasfl) [inline README.md]: Small typo. Should be:
```
   execute a shell command
```

Comment (AlexWayfer) [inline README.md]: Fixed!

**Comment by AlexWayfer** [comment]
Oh, we have failing builds with `jruby` :scream:

**Comment by thomasfl** [comment]
Do you know why it fails? 😕

**Comment by AlexWayfer** [comment]
@thomasfl JRuby updates file content correct, but doesn't update `File.mtime` :thinking: 

```ruby
before = File.stat(filename).mtime
File.write(filename, 'content3')
File.read(filename) # => "content3"
after = File.stat(filename).mtime
before == after # => true
```

**Comment by AlexWayfer** [comment]
`Time#to_f`

`jruby-9.1.7.0`:
```
1488447511.0
1488447511.0
```

`mri-2.4.0`:
```
1488447533.3233862
1488447533.5233934
```

**Comment by AlexWayfer** [comment]
Fixed in https://github.com/thomasfl/filewatcher/pull/54/commits/f19b601c36144a361669e7f362b44735270b0878

**Pull request merged**